src/System.Reflection.Metadata/src/System/Reflection/Metadata/Decoding/SignatureDecoder.cs

```diff
@@ -17,7 +16,8 @@ public static class SignatureDecoder
         /// <param name="handle">A type definition, reference, or specification handle.</param>
         /// <param name="provider">The type provider.</param>
         /// <exception cref="System.BadImageFormatException">The handle does not represent a valid type reference, definition, or specification.</exception>
-        public static TType DecodeType<TType>(Handle handle, ISignatureTypeProvider<TType> provider)
+        public static TType DecodeType<TType, TProvider>(Handle handle, TProvider provider)
```

Comment:
We pass this provider around liberally. Should it be passed by ref everywhere if we now think it should be a struct?